Mg streams improvements (#168)
memgraph/docs: docs/reference-guide/streams/overview.md

> 

Reviewer: Maybe mention `at-least once` semantics?

## Start a stream

```cypher
START STREAM <stream name>;
START ALL STREAMS;
```
Starts a stream (or all streams) with name `<stream name>`.

When a stream is started, it should resume from the last committed offset. If
there is no committed offset for the consumer group, then the largest offset
will be used, therefore only the new messages will be consumed.

### At least once semantics

In stream processing it is important to have some guarantees about how failures

> are handled. In case connecting external application such as Memgraph to Kafka

Reviewer 2: ```suggestion
are handled. When connecting an external application such as Memgraph to a Kafka stream,
```

> streams, there are two possible ways to handle failures during message

Reviewer 2: ```suggestion
 there are two possible ways to handle failures during message
```

Author: I did your proposed changes but wrapped the lines differently to remain into the 80 char limit.

processing:
1. Every message is processed **at least once**: the message offsets are
committed to the Kafka cluster after the processing is done. This means if the

> committing failes, then the messages can get processed multiple times.

Reviewer 2: ```suggestion
committing fails, then the messages can get processed multiple times.
```

there is no committed offset for the consumer group, then the largest offset
will be used, therefore only the new messages will be consumed.

### At least once semantics

In stream processing it is important to have some guarantees about how failures
are handled. In case connecting external application such as Memgraph to Kafka
streams, there are two possible ways to handle failures during message
processing:
1. Every message is processed **at least once**: the message offsets are
committed to the Kafka cluster after the processing is done. This means if the
committing failes, then the messages can get processed multiple times.
2. Every message is precessed **at most once**: the message offsets are
commited to the Kafka cluster right after they are recieved, before the
processing is started. This means if the processing fails, then the same
messages won't be processed again.

Missing a message can result in missing an edge that would connect two

> independent components of the graph. Therefore in our understanding missing
> some information is a bigger problem for graphs than having some information
> duplicated, so we implemented our streams using the **at least once**
> semantics: for every batch of messages the queries returned by the
> transformations are executed and committed to the database before committing
> the message offset to the Kafka cluster. However we cannot guarantee **exactly

Reviewer: ```suggestion
independent components of the graph. Therefore, we think that missing
some information is a bigger problem for graphs than having some information
duplicated, so we implemented our streams using the **at least once**
semantics, i.e. for every batch of messages the queries returned by the
transformations are executed and committed to the database before committing
the message offset to the Kafka cluster. However, we cannot guarantee **exactly
```
